Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -126,1402 +117,54 @@ namespace CSSPropertyParserHelpers {
-    //
-    // see https://www.w3.org/TR/2011/WD-css3-images-20110217/#linear-gradients.
-
-    static constexpr std::pair<CSSValueID, CSSPrefixedLinearGradientValue::Vertical> verticalMappings[] {
-        { CSSValueTop, CSSPrefixedLinearGradientValue::Vertical::Top },
-        { CSSValueBottom, CSSPrefixedLinearGradientValue::Vertical::Bottom },
-    };
-    static constexpr SortedArrayMap verticalMap { verticalMappings };
-
-    static constexpr std::pair<CSSValueID, CSSPrefixedLinearGradientValue::Horizontal> horizontalMappings[] {
-        { CSSValueLeft, CSSPrefixedLinearGradientValue::Horizontal::Left },
-        { CSSValueRight, CSSPrefixedLinearGradientValue::Horizontal::Right },
-    };
-    static constexpr SortedArrayMap horizontalMap { horizontalMappings };
-
-    auto consumeKeywordGradientLineKnownHorizontal = [&](CSSParserTokenRange& range, CSSPrefixedLinearGradientValue::Horizontal knownHorizontal) -> CSSPrefixedLinearGradientValue::GradientLine {
-        if (auto vertical = consumeIdentUsingMapping(range, verticalMap))
-            return std::make_pair(knownHorizontal, *vertical);
-        return knownHorizontal;
-    };
-
-    auto consumeKeywordGradientLineKnownVertical = [&](CSSParserTokenRange& range, CSSPrefixedLinearGradientValue::Vertical knownVertical) -> CSSPrefixedLinearGradientValue::GradientLine {
-        if (auto horizontal = consumeIdentUsingMapping(range, horizontalMap))
-            return std::make_pair(*horizontal, knownVertical);
-        return knownVertical;
-    };
-
-    auto consumeKeywordGradientLine = [&](CSSParserTokenRange& range) -> std::optional<CSSPrefixedLinearGradientValue::GradientLine> {
-        switch (range.peek().id()) {
-        case CSSValueLeft:
-            range.consumeIncludingWhitespace();
-            return consumeKeywordGradientLineKnownHorizontal(range, CSSPrefixedLinearGradientValue::Horizontal::Left);
-        case CSSValueRight:
-            range.consumeIncludingWhitespace();
-            return consumeKeywordGradientLineKnownHorizontal(range, CSSPrefixedLinearGradientValue::Horizontal::Right);
-        case CSSValueTop:
-            range.consumeIncludingWhitespace();
-            return consumeKeywordGradientLineKnownVertical(range, CSSPrefixedLinearGradientValue::Vertical::Top);
-        case CSSValueBottom:
-            range.consumeIncludingWhitespace();
-            return consumeKeywordGradientLineKnownVertical(range, CSSPrefixedLinearGradientValue::Vertical::Bottom);
-        default:
-            return { };
-        }
-    };
-
-    std::optional<CSSPrefixedLinearGradientValue::GradientLine> gradientLine;
-
-    const auto angleConsumeOptions = CSSPropertyParserOptions {
-        .parserMode = context.mode,
-        .unitless = UnitlessQuirk::Forbid,
-        .unitlessZero = UnitlessZeroQuirk::Allow
+#if !ENABLE(VARIATION_FONTS)
+    auto isIntegerAndDivisibleBy100 = [](double value) {
+        ASSERT(value >= 1 && value <= 1000);
```

Comment:
I'd break this into two separate assertions, not sure other people agree, though.